beacon-chain/rpc/validator/status.go

```diff
@@ -88,6 +88,90 @@ func (vs *Server) MultipleValidatorStatus(
 	}, nil
 }
 
+func (vs *Server) CheckDoppelGanger(ctx context.Context, req *ethpb.DoppelGangerRequest) (*ethpb.DoppelGangerResponse, error) {
+	if vs.SyncChecker.Syncing() {
+		return nil, status.Errorf(codes.Unavailable, "Syncing to latest head, not ready to respond")
+	}
+	headState, err := vs.HeadFetcher.HeadState(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Internal, "Could not get head state")
+	}
+	// We walk back from the current head to the previous 2 states. With S_i , i := 0,1,2. i = 0
+	// would signify the current head state in this epoch.
+	currEpoch := helpers.SlotToEpoch(headState.Slot())
+	previousEpoch, err := currEpoch.SafeSub(1)
+	if err != nil {
+		previousEpoch = currEpoch
+	}
+	olderEpoch, err := previousEpoch.SafeSub(1)
+	if err != nil {
+		olderEpoch = previousEpoch
+	}
+	prevState, err := vs.StateGen.StateBySlot(ctx, params.BeaconConfig().SlotsPerEpoch.Mul(uint64(previousEpoch)))
+	if err != nil {
+		return nil, status.Error(codes.Internal, "Could not get previous state")
+	}
+	olderState, err := vs.StateGen.StateBySlot(ctx, params.BeaconConfig().SlotsPerEpoch.Mul(uint64(olderEpoch)))
+	if err != nil {
+		return nil, status.Error(codes.Internal, "Could not get previous state")
+	}
+	resp := &ethpb.DoppelGangerResponse{
+		Responses: []*ethpb.DoppelGangerResponse_ValidatorResponse{},
+	}
+	for _, v := range req.ValidatorRequests {
+		// If the validator's last recorded epoch was
+		// less than 2 epoch ago, this method will not
+		// be able to catch duplicates.
+		if v.Epoch+2 > currEpoch {
```

Comment:
```suggestion
		if v.Epoch+2 >= currEpoch {
```
Rationale: if the validator attested in `olderEpoch` and the attestation was included in `previousEpoch` the balance change will not be reflected until the end of `currentEpoch`.